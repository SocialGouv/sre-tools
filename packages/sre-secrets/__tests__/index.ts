const fs = require("fs")
const util = require("util")
const yaml = require("js-yaml")
const exec = util.promisify(require("child_process").exec)

describe("Test sealed secrets generation", () => {
  const filePath = "./__tests__/data/.secrets.yaml"
  const folderPath = `${process.env.RUNNER_TEMP || "/tmp"}/sre-secrets`
  const matchers = {
    spec: {
      encryptedData: {
        toto: expect.any(String),
        tata: expect.any(String),
      },
    },
  }

  test("Generate sealed secrets", async () => {
    const cmd = `node ./dist/index.js --from=${filePath} --to=${folderPath}`
    const { stdout, stderr } = await exec(cmd)
    if (stdout) console.log("stdout:", stdout)
    if (stderr) console.log("stderr:", stderr)
  })

  test("Check dev snapshot", () => {
    const path = `${folderPath}/environments/dev/app.sealed-secret.yaml`
    const content = yaml.safeLoad(fs.readFileSync(path, "utf8"))
    expect(content).toMatchSnapshot(matchers)
  })

  test("Check preprod snapshot", () => {
    const path = `${folderPath}/environments/preprod/app.sealed-secret.yaml`
    const content = yaml.safeLoad(fs.readFileSync(path, "utf8"))
    expect(content).toMatchSnapshot(matchers)
  })

  test("Check prod snapshot", () => {
    const path = `${folderPath}/environments/prod/app.sealed-secret.yaml`
    const content = yaml.safeLoad(fs.readFileSync(path, "utf8"))
    expect(content).toMatchSnapshot(matchers)
  })
})
