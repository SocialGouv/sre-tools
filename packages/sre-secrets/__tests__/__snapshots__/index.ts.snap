// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test sealed secrets generation Check dev snapshot 1`] = `
Object {
  "apiVersion": "bitnami.com/v1alpha1",
  "kind": "SealedSecret",
  "metadata": Object {
    "annotations": Object {
      "sealedsecrets.bitnami.com/cluster-wide": "true",
    },
    "name": "app-sealed-secret",
    "namespace": "my-app-namespace",
  },
  "spec": Object {
    "encryptedData": Object {
      "tata": Any<String>,
      "toto": Any<String>,
    },
    "template": Object {
      "metadata": Object {
        "annotations": Object {
          "sealedsecrets.bitnami.com/cluster-wide": "true",
        },
        "name": "app-sealed-secret",
      },
      "type": "Opaque",
    },
  },
}
`;

exports[`Test sealed secrets generation Check preprod snapshot 1`] = `
Object {
  "apiVersion": "bitnami.com/v1alpha1",
  "kind": "SealedSecret",
  "metadata": Object {
    "annotations": Object {
      "sealedsecrets.bitnami.com/cluster-wide": "true",
    },
    "name": "app-sealed-secret",
    "namespace": "my-app-namespace",
  },
  "spec": Object {
    "encryptedData": Object {
      "tata": Any<String>,
      "toto": Any<String>,
    },
    "template": Object {
      "metadata": Object {
        "annotations": Object {
          "sealedsecrets.bitnami.com/cluster-wide": "true",
        },
        "name": "app-sealed-secret",
      },
      "type": "Opaque",
    },
  },
}
`;

exports[`Test sealed secrets generation Check prod snapshot 1`] = `
Object {
  "apiVersion": "bitnami.com/v1alpha1",
  "kind": "SealedSecret",
  "metadata": Object {
    "annotations": Object {},
    "name": "app-sealed-secret",
    "namespace": "my-app-namespace",
  },
  "spec": Object {
    "encryptedData": Object {
      "tata": Any<String>,
      "toto": Any<String>,
    },
    "template": Object {
      "metadata": Object {
        "annotations": Object {},
        "name": "app-sealed-secret",
      },
      "type": "Opaque",
    },
  },
}
`;
